src/util/InputRangeUtils.h

```diff
@@ -225,4 +240,30 @@ template <typename Range, typename F>
 CachingContinuableTransformInputRange(Range&&, F)
     -> CachingContinuableTransformInputRange<all_t<Range>, F>;
 
+CPP_class_template(typename ViewType)(
+    requires(ql::ranges::input_range<ViewType>)) struct ConcatenatedInputRange
+    : ad_utility::InputRangeFromGet<ql::ranges::range_value_t<ViewType>> {
+  using ViewCollection = std::vector<std::unique_ptr<ViewType>>;
+  ViewCollection viewCollection_;
+  ql::ranges::iterator_t<ViewCollection> collectionIt_;
+
+  ConcatenatedInputRange(ViewCollection viewCollection)
+      : viewCollection_(std::move(viewCollection)),
+        collectionIt_(viewCollection_.begin()) {}
+
+  std::optional<ql::ranges::range_value_t<ViewType>> get() {
+    while (collectionIt_ != viewCollection_.end()) {
+      auto elementOpt = (*collectionIt_)->get();
+      if (elementOpt.has_value()) {
+        // Element exists in the currently iterated view
+        return elementOpt;
+      }
+      // Done with current view, but not all views
+      collectionIt_++;
+    }
+    // Done iterating over all views
+    return std::nullopt;
+  }
+};
```

Comment:
I like this very much as a simple way to implement this at the cost of additional indirections (because of the vector<unique_ptr>, but that is not an issue when rewriting generators which are inherently inefficient (on the iteration level) anyway.